### PR TITLE
Implemented S3DistCpActivity

### PR DESCRIPTION
**Pull request by sethyates**

### CODE REVIEW

**Comment by ysalomon** [comment]
LGTM but Travis seems unhappy

**Comment by sethyates** [comment]
Stinking Scala 2.10 lower limitation on case class members.  I'll convert to a regular class.

File: src/main/scala/com/krux/hyperion/activity/S3DistCpActivity.scala [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    endpoint: Option[String] = endpoint,
+    storageClass: Option[StorageClass] = storageClass,
+    sourcePrefixesFile: Option[String] = sourcePrefixesFile
+  ) = new S3DistCpActivity(id, runsOn, dependsOn, preconditions, onFailAlarms, onSuccessAlarms, onLateActionAlarms,
+    source, dest, sourcePattern, groupBy, targetSize, appendLastToFile, outputCodec, s3ServerSideEncryption,
+    deleteOnSuccess, disableMultipartUpload, chunkSize, numberFiles, startingIndex, outputManifest,
+    previousManifest, requirePreviousManifest, copyFromManifest, endpoint, storageClass, sourcePrefixesFile)
+
+  def named(name: String) = this.copy(id = PipelineObjectId.withName(name, id))
+  def groupedBy(group: String) = this.copy(id = PipelineObjectId.withGroup(group, id))
```
Comment (ysalomon) [inline src/main/scala/com/krux/hyperion/activity/S3DistCpActivity.scala]: actually, should this be private?

Comment (sethyates) [inline src/main/scala/com/krux/hyperion/activity/S3DistCpActivity.scala]: Should what be private?

**Pull request closed without merging**